### DISCUSSION TITLE
Filter from a calculated column

### DISCUSSION THREAD

**Discussion by mbh00** [filamentphp/filament#11673] [Help]
### Package

Table builder

### Package Version

v3.2.34

### How can we help you?

I have a table with a calculated column as stated here 
https://filamentphp.com/docs/3.x/tables/columns/getting-started#calculated-state

How can I create a filter based on that column, for example using the QueryBuilder filter and a number constraint?
this is the table 
public function table(Table $table): Table
    {
       /***
performing some calculations here to produce $res variable
       **/
        return $table->paginated([5,10,25,50])->queryStringIdentifier('us2')
        ->query(Exam::with('certRel')->where('from',$this->record->id)->orWhereRelation('users', 'user', $this->record->id)->latest('added_at'))
            ->columns([
                Tables\Columns\TextColumn::make('certRel.name')->sortable()->searchable()->label('Title')
                ->description(fn (Exam $record): string => $record->name),
                Tables\Columns\TextColumn::make('name')->sortable()->searchable()->label('Code')->toggleable(isToggledHiddenByDefault: true),
                Tables\Columns\TextColumn::make('type')->badge(),
                Tables\Columns\TextColumn::make('result')->label('% Perf.')->sortable()->badge()
                ->state(fn (Exam $record) => $res[$record->id]??0)
                ->color(fn ($record): string =>isset($res[$record->id])? 'success':'gray')
                ->formatStateUsing(fn ($state,$record): string => isset($res[$record->id])? $state.'%':'12%'),
            Tables\Columns\TextColumn::make('quest')->label('Questions')->sortable(),
            Tables\Columns\TextColumn::make('timer')->label('Timer')->sortable(),
            ])
            ->filters([
                Tables\Filters\SelectFilter::make('type')->label('Type')
                ->options(['0' => 'Test','1' => 'Exam']),
                ]);
    }

Here I want to add a filter based on the 'result' calculated column. 
When I put something like Filter::make('result'), there is an SQL error saying column 'result' is not found. This means that the query is done in the database. But I need the filter to search through the actual records displayed in the table, since the calculated column is there, not in the database. 

How can I handle this?

**Comment by leandrocfe** [comment]
Maybe a [custom filter](https://filamentphp.com/docs/3.x/tables/filters/custom)?

**Reply by mbh00** [reply]
I tried the custom filter, but it is still querying the database.

**Reply by leandrocfe** [reply]
This will use the database. You should apply in the `query` method, the same logic that you are using in the calculated column, to filter it.

you could also use a virtual column in your database.

**Reply by mbh00** [reply]
The calculation is too complex and involves data from different sources. Putting that in a query is simply impossible.

**Comment by mbh00** [comment]
Maybe a feature request?
